docs: make setting `gas` optional and require evmVersion (#1496)
hyperledger/besu-docs: versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md

   gasPrice: 0;
   ```

> 2. Set the gas limit for a transaction (that is, contract creation) to be the block gas limit - 1.

Reviewer: ```suggestion
1. Set the gas limit for a transaction (that is, contract creation) to be the block gas limit - 1.
```

Reviewer: Markdown should auto increment the numbering

   ```js
   gas: "0x1ffffffffffffe";
   ```
   

> > In most deployments, setting the `gasPrice` to 0 is enough (if contract is not that big).

Reviewer: ```suggestion
      > For most deployments, setting `gasPrice` to `0` should cover transaction costs, especially if the contract is not large.
```

> 3. Specify the `evmVersion` when using latest Solidity versions:

Reviewer: ```suggestion
3. Specify `evmVersion` when using the latest Solidity version:
```

   ```js
   solidity: {
      version: "0.8.20",
      settings: {

>         evmVersion: "london", // required for besu

Reviewer: ```suggestion
        evmVersion: "london", // required for Besu
```
